upload-api/stores/agent/store.js

```diff
@@ -328,12 +378,31 @@ const toIndexEntry = (path) => {
  */
 const list = async (connection, { prefix, suffix }) => {
   try {
+    // attempt to read first from dynamo
+    const dynamoCommand = new GetItemCommand({
+      TableName: connection.tables.index.name,
+      Key: {
+        "task": {
+          S: prefix
```

Comment:
To remove the trailing slash:

```suggestion
          S: prefix.slice(0, -1)
```